bothan-core/src/manager/crypto_asset_info/price/tasks.rs

```diff
@@ -75,24 +70,20 @@ where
         .collect()
 }
 
-async fn compute_signal_result<S, W>(
+async fn compute_signal_result<S: Store>(
     id: &str,
-    workers: &HashMap<String, W>,
+    store: &S,
     registry: &Registry<Valid>,
     stale_cutoff: i64,
     cache: &PriceCache<String>,
     records: &mut Vec<PriceSignalComputationRecord>,
-) -> Result<Decimal, Error>
-where
-    S: Store + 'static,
-    W: AssetWorker<S>,
-{
+) -> Result<Decimal, Error> {
     match registry.get(id) {
         Some(signal) => {
             let mut record = SignalComputationRecord::new(id.to_string());
```

Comment:
This is your intention to push 2 records of the same symbol? (first will push when require pre-req and push success on other record right?)

Reply:
Also I think no record has been tested in test na